tests/src/test/java/alluxio/client/rest/TestCase.java

```diff
@@ -29,52 +29,41 @@
 import java.net.URL;
 import java.util.Map;
 import javax.annotation.concurrent.NotThreadSafe;
+import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
 /**
  * Represents a REST API test case.
  */
 @NotThreadSafe
 public final class TestCase {
-  private String mHostname;
-  private int mPort;
-  private String mEndpoint;
-  private Map<String, String> mParameters;
-  private String mMethod;
-  private Object mExpectedResult;
-  private TestCaseOptions mOptions;
-  private String mPrefix;
+  /* Required */
+  private final String mHostname;
+  private final int mPort;
+  private final String mEndpoint;
+  private final Map<String, String> mParameters;
+  private final String mMethod;
+  private final Object mExpectedResult;
+  /* Optional */
+  private final TestCaseOptions mOptions;
+  private final String mServicePrefix;
```

Comment:
Personally I'd prefer if `mServicePrefix` was not optional, and instead the callers who did not have a prefix just pass in an empty string. I don't know if we'd prefer that over the more "Java-native" approach with (# optional parameters)^2 constructors.

Reply:
If `""` is a valid prefix, it is fine to use that instead of optional. I think here that's the case. If you find yourself in the situation that you have to check whether a string is `""` constantly (instead of checking for null), then `Optional` is definitely better.

Reply:
Decided to axe the entire prefix thing and use `mBaseUri` instead. There are no longer any optional parameters to this class either, as `mExpectedResult` was not used anywhere except for one function.